ide/nativeimage.api/src/org/netbeans/modules/nativeimage/api/debug/NIDebuggerService.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.nativeimage.api.debug;
+
+import org.netbeans.api.annotations.common.CheckForNull;
+import org.netbeans.api.project.Project;
+import org.netbeans.modules.nativeimage.api.NIUtils;
+import org.netbeans.modules.nativeimage.spi.debug.NIDebuggerProvider;
+import org.netbeans.modules.nativeimage.spi.debug.NIDebuggerServiceProvider;
+import org.openide.filesystems.FileObject;
+import org.openide.util.Lookup;
+
+/**
+ * A service that provides native image debugger.
+ */
+public final class NIDebuggerService {
```

Comment:
Why is this class in API at all? It's onlyl purpose seems to be the create the `NIDebugger` can't the necessary methods be added to the `NIDebugger` as `static NIDebugger create(...)`? The `NIDebuggerService` could be hidden from the API or be removed completely.

Reply:
Removed.